core/PruManager.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * PruManager.cpp
+ * This code currently only works on BBAI
+ *
+ *	Created on: July 3, 2021
+ *		Author: Dhruva Gole
+ */
+
+#include "../include/PruManager.h"
+#include "MiscUtilities.h"
+#include <iostream>
+
+#if ENABLE_PRU_RPROC == 1
+PruManagerRprocMmap::PruManagerRprocMmap(unsigned int pruNum, unsigned int v)
+{
+	/* constructor for initializing the necessary path variables
+	 * based on the value of pru_num to choose:
+	 * 0 for PRU1 core 0
+	 * 1 for PRU1 core 1
+	 * 2 for PRU2 core 0
+	 * 3 for PRU2 core 1
+	 */
+	pru_num = pruNum;
+	verbose = v;
+	unsigned int pruss = pru_num / 2 + 1;
+	unsigned int prucore = pru_num % 2;
+
+	basePath = "/dev/remoteproc/pruss" + std::to_string(pruss) + "-core" + std::to_string(prucore) + "/";
+	statePath = basePath + "state";
+	firmwarePath = basePath + "firmware";
+	firmware = "am57xx-pru" + std::to_string(pruss) + "_" + std::to_string(prucore) + "-fw";
+	firmwareCopyCommand = "sudo rm /lib/firmware/" + firmware + ";sudo ln -s /home/debian/Bela-dhruva/pru/blinkR30.pru1_0.out /lib/firmware/" + firmware; // **NOTE**: Change the name of .out file according afterwards here and below cases.
+	// 0 : pruss1-core 0 in AI -> 4b234000
+	// 1 : pruss1-core 1 in AI -> 4b238000
+	// 2 : pruss2-core 0 in AI -> 4b2b4000
+	// 3 : pruss2-core 1 in AI -> 4b2b8000
+	//
+	// 0 : pruss-core 0 in BBB -> 4a334000
+	// 1 : pruss-core 1 in BBB -> 4a338000
+
+#ifdef IS_AM572x	// base addresses for BBAI
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(0,0x4b234000));
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(1,0x4b238000));
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(2,0x4b2b4000));
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(3,0x4b2b8000));
+#else	// base addresses for BBB
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(0,0x4a334000));
+	pru_addr.insert(std::pair<unsigned int, unsigned int>(1,0x4a338000));
+#endif
+	long addr2 = 0x4b200000;
```

Comment:
this is a local variable. Remove `long`

Reply:
also, its assignment should also be part of the `#ifdef`.

Reply:
(unless it's the same for AM572x and AM335x)

Reply:
I feel like we discussed this already ...

also, rename it from `addr2` to something more meaningful.

Reply:
Yes, this is still in my to-do list...

Reply:
done